grails-app/services/org/pih/warehouse/data/LocationDataService.groovy

```diff
@@ -115,12 +129,20 @@ class LocationDataService {
         location.name = params.name
         location.active = Boolean.valueOf(params.active)
         location.locationNumber = params.locationNumber
-        location.locationType = params.locationType ? LocationType.findByNameLike(params.locationType + "%") : null
         location.locationGroup = params.locationGroup ? LocationGroup.findByName(params.locationGroup) : null
         location.parentLocation = params.parentLocation ? Location.findByNameOrLocationNumber(params.parentLocation, params.parentLocation) : null
         location.organization = params.organization ? Organization.findByCodeOrName(params.organization, params.organization) : null
         location.address = createOrUpdateAddress(params, location?.address?.id)
 
+        def locationType = params.locationType ? LocationType.findByNameLike(params.locationType + "%") : null
+        def currentLocationType = location.locationType
+
+        //Do not allow to change internal type location to non-internal
+        if (((currentLocationType?.isInternalLocation() || currentLocationType?.isZone()) && (locationType?.isInternalLocation() || locationType?.isZone())) || !(currentLocationType?.isInternalLocation() || currentLocationType?.isZone())) {
+            location.locationType = locationType
+            location.supportedActivities = currentLocationType && locationType == currentLocationType ? location.supportedActivities : new HashSet(location.locationType.getSupportedActivities())
```

Comment:
If we change location types then we should only clear the Location's supported activities. In other words, do not add the location type's supported activities because the location will inherit these by default if it's supported actitivies association is empty.